mods/gen2/random-teams.js

```diff
@@ -91,13 +91,12 @@ class RandomGen2Teams extends RandomGen3Teams {
 
 			// Moves counter.
 			restrictMoves = set.other.restrictMoves;
-			let moves = set.moveset.moves;
-			for (let i = 0; i < moves.length; i++) {
-				if (restrictMoves[moves[i]]) restrictMoves[moves[i]]--;
-				if (restrictMoves['phazing'] && (moves[i] === "roar" || moves[i] === "whirlwind")) {
+			for (const move of set.moveset.moves) {
```

Comment:
Would use `moveid` here. I'm trying to be consistent by using `moveid` for move IDs and `move` for move objects.